lib/IRGen/GenCall.cpp

```diff
@@ -1961,23 +1966,24 @@ bool irgen::addNativeArgument(IRGenFunction &IGF, Explosion &in,
 
 /// Emit a direct parameter that was passed under a C-based CC.
 static void emitDirectForeignParameter(IRGenFunction &IGF,
-                                       Explosion &in,
-                                       llvm::Type *coercionTy,
-                                       Explosion &out,
+                                       const clang::CodeGen::ABIArgInfo &AI,
+                                       Explosion &in, Explosion &out,
```

Comment:
Same comment about reordering.  The order was "source value, source foreign info, target value, target type, target type info", and now it's rather muddled.

Reply:
Swapped the `AI` for the `coercionTy`.